yt/fields/magnetic_field.py

```diff
@@ -210,12 +260,5 @@ def setup_magnetic_field_aliases(registry, ds_ftype, ds_fields, ftype="gas"):
         convert = lambda x: x.in_units(to_units)
     else:
         convert = lambda x: x.to_equivalent(to_units, equiv)
-    def mag_field(fd):
-        def _mag_field(field, data):
-            return convert(data[fd])
-        return _mag_field
-    for ax, fd in zip(registry.ds.coordinates.axis_order, ds_fields):
-        registry.add_field((ftype,"magnetic_field_%s" % ax),
-                           sampling_type="cell", 
-                           function=mag_field(fd),
-                           units=unit_system[to_units.dimensions])
```

Comment:
I don't quite understand the reason to use `unit_system[to_units.dimensions]` instead of `to_units` here. Could someone explain?

Reply:
Actually, I have kept this in the refactored code. That's why the tests pass.